contrib/ruby_event_store-outbox/lib/ruby_event_store/outbox/tempo.rb

```diff
@@ -0,0 +1,21 @@
+module RubyEventStore
+  module Outbox
+    class Tempo
+      def initialize(max_batch_size)
+        raise ArgumentError if max_batch_size < 1
+        @max_batch_size = max_batch_size
+      end
+
+      def batch_size
+        @batch_size = next_batch_size
+      end
+
+      private
+
+      def next_batch_size
+        return 1 if @batch_size.nil?
+        [@batch_size * 2, @max_batch_size].min
```

Comment:
Maybe the multiplier value could be also configurable, or this is just making this more complex than needed? 
2 is good default  :)

Reply:
Nah, I will make it a constant, but the key is to:
- start with 1
- go quickly (exponentially) to the desired batch size, with exponential, 2 or 3 doesn't make almost any difference